### PR TITLE
Adding test to db quries

### LINKED ISSUE #58: Add unit tests for query logic
## Description:
- Investigate best way to mock controllers and database contents
- References #45

### PR DESCRIPTION
**Pull request by tobiaswright**
## Type of PR

- [ ] Documentation changes
- [ ] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
Added test to db quries

## Review notes
Test does not cover 100%, blockers here are around async calls and private methods and variables that make the public functions not testable

## Issues Closed or Referenced

- Closes #58  (this will automatically close the issue when the PR closes)
- References #<issue number> (this references the issue but does not close with PR)

### CODE REVIEW

File: src/db/__tests__/query.tests.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        expect(query.queryActors({})).resolves.toEqual(["Lawrence Fishburn"]);
+    });
+
+    describe("Expected SQL statement given the argument", () => {
+
+        const query = new CosmosDBProvider("http://www.db.com", "dbkey", "db", "collection", null);
+        query.queryDocuments = jest.fn();
+
+        test("Defualt params sent to DB", () => {
+            query.queryActors({});
```
Comment (aflinchb) [inline src/db/__tests__/query.tests.ts]: typo on line 26 - should be test("Default ..."

File: src/db/__tests__/query.tests.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        test("Params with pageNumber < 0", () => {
+            query.queryMovies({pageSize: 1001, pageNumber: -20});
+            const call = "select m.id, m.partitionKey, m.movieId, m.type, m.textSearch, m.title, m.year, m.runtime, m.rating, m.votes, m.totalScore, m.genres, m.roles from m where m.type = 'Movie'  order by m.title offset 0 limit 1000 ";
+
+            expect(query.queryDocuments).toBeCalledWith(call);
+        });
+
+        test("Params with a query", () => {
+            query.queryMovies({q: "The Squel"});
+            const call = "select m.id, m.partitionKey, m.movieId, m.type, m.textSearch, m.title, m.year, m.runtime, m.rating, m.votes, m.totalScore, m.genres, m.roles from m where m.type = 'Movie'  and contains(m.textSearch, 'the squel')  order by m.title offset 0 limit 100 ";
```
Comment (aflinchb) [inline src/db/__tests__/query.tests.ts]: I know it doesn't matter functionality wise -- but small typo here :) should be "The Sequel"

File: src/db/__tests__/query.tests.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        });
+
+        test("Params with a rating", () => {
+            query.queryMovies({rating: 5});
+            const call = "select m.id, m.partitionKey, m.movieId, m.type, m.textSearch, m.title, m.year, m.runtime, m.rating, m.votes, m.totalScore, m.genres, m.roles from m where m.type = 'Movie'  and m.rating >= 5  order by m.title offset 0 limit 100 ";
+
+            expect(query.queryDocuments).toBeCalledWith(call);
+        });
+
+        test("Params with a toprated", () => {
```
Comment (bartr) [inline src/db/__tests__/query.tests.ts]: Need to remove topRated per design review

Comment (tobiaswright) [reply src/db/__tests__/query.tests.ts]: done!

File: src/db/__tests__/query.tests.ts (hunk 2 above)
Comment (tobiaswright) [reply src/db/__tests__/query.tests.ts]: done!

File: src/db/__tests__/query.tests.ts (hunk 1 above)
Comment (tobiaswright) [reply src/db/__tests__/query.tests.ts]: done!

**Comment by tobiaswright** [comment]
Here is the current coverage test. Other changes have been made. Will commit andmerge when I land. @aflinchb 
Screenshot from 2020-03-02 21-48-44

**Comment by tobiaswright** [comment]
In light of 68% coverage for queries, should we attempt to get the percentage up before we merge?

**Comment by aflinchb** [comment]
> In light of 68% coverage for queries, should we attempt to get the percentage up before we merge?

I think we should go ahead and merge this one, but create a new issue to cover the work of deciding w/ the team if this is sufficient coverage, define what sufficient coverage is for all languages (we should do this regardless), and/or investigating/implemented the needed changes to get the coverage up.

**Pull request merged**